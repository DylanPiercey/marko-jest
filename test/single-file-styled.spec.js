const path = require('path');
const { initComponent, createTestSandbox } = require('../test-utils');

describe('single-file with styled component', () => {
  const componentClass = initComponent(path.resolve(__dirname, './resources/single-file-styled/index.marko'));

  let testSandbox;
  let component;

  beforeEach(() => {
    testSandbox = createTestSandbox();
  });

  afterEach(() => {
    testSandbox.reset();
  });

  describe('on rendering', () => {
    beforeEach(async () => {
      component = await testSandbox.renderComponent(componentClass, {});
    });

    it('should render correctly', () => {
      expect(component.els).toMatchSnapshot();
    });
  });

  describe('on triggering click handler', () => {
    beforeEach(async () => {
      component = await testSandbox.renderComponent(componentClass, {});

      component.el.click();
      component.update();
    });

    it('should update the element', () => {
      expect(component.els).toMatchSnapshot();
    });
  });
});
