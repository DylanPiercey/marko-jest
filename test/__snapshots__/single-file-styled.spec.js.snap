// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single-file with styled component on rendering should render correctly 1`] = `
Array [
  <button
    class="btn"
  >
    <span>
      Click me
    </span>
  </button>,
]
`;

exports[`single-file with styled component on triggering click handler should update the element 1`] = `
Array [
  <button
    class="btn toggled"
  >
    <span>
      DONE
    </span>
  </button>,
]
`;
